add-friend.php

```diff
@@ -1,6 +1,17 @@
 <?php
-	require "db.php";
-	// TODO: Test
-	$db->query("INSERT INTO friend (usename, friend) VALUES" [$_SESSION["username"], $_POST["friend"]]);
-	header("Location: home.php");
+	require_once "db.php";
+
+	$error = false;
+	if (isset($_POST["friend"])) {
+		$results = $db->query("select * from user where username = ?", [$_POST["friend"]]);
+		if (count($results) == 0 || $results[0]["username"] == $_SESSION["username"]) {
+			$error = "Username not found";
+		} else {
+			$db->query("INSERT INTO friendship (user, friend) VALUES (?, ?)", [$_SESSION["username"], $_POST["friend"]]);
+			header("Location: home.php");
+			return;
+		}
+	}
+
+	header("Location: home.php?error=" . $error . "#add-friend");
```

Comment:
I think I know a better way we could handle errors actually. If we put them in $_SESSION before redirecting, then when the following page is loaded, PHP could check if the session has an error, and if so print it *and* clear the error. Then if the user refreshes they wouldn't see the error again.